provisioner/zalando_eks.go

```diff
@@ -0,0 +1,181 @@
+// EKS clusters by passing its location and and IAM role to use.
+func NewZalandoEKSProvisioner(
+	execManager *command.ExecManager,
+	secretDecrypter decrypter.Decrypter,
+	assumedRole string,
+	awsConfig *aws.Config,
+	options *Options,
+) Provisioner {
+	provisioner := &ZalandoEKSProvisioner{
+		clusterpyProvisioner: clusterpyProvisioner{
+			awsConfig:         awsConfig,
+			assumedRole:       assumedRole,
+			execManager:       execManager,
+			secretDecrypter:   secretDecrypter,
+			manageMasterNodes: false,
+			manageEtcdStack:   false,
+		},
+	}
+
+	if options != nil {
+		provisioner.dryRun = options.DryRun
+		provisioner.applyOnly = options.ApplyOnly
+		provisioner.updateStrategy = options.UpdateStrategy
+		provisioner.removeVolumes = options.RemoveVolumes
+		provisioner.modifier = options.Modifier
+	}
+
+	return provisioner
+}
+
+func (z *ZalandoEKSProvisioner) Supports(cluster *api.Cluster) bool {
+	return cluster.Provider == string(ZalandoEKSProvider)
+}
+
+func (z *ZalandoEKSProvisioner) Provision(
+	ctx context.Context,
+	logger *log.Entry,
+	cluster *api.Cluster,
+	channelConfig channel.Config,
+) error {
+	if !z.Supports(cluster) {
+		return ErrProviderNotSupported
+	}
+
+	awsAdapter, err := z.setupAWSAdapter(logger, cluster)
+	if err != nil {
+		return fmt.Errorf("failed to setup AWS Adapter: %v", err)
+	}
+
+	eksTokenSource := eks.NewTokenSource(awsAdapter.session, eksID(cluster.ID))
+
+	logger.Infof(
+		"clusterpy: Prepare for provisioning EKS cluster %s (%s)..",
+		cluster.ID,
+		cluster.LifecycleStatus,
+	)
+
+	return z.provision(
+		ctx,
+		logger,
+		awsAdapter,
+		eksTokenSource,
+		cluster,
+		channelConfig,
+	)
+}
+
+func (z *ZalandoEKSProvisioner) Decommission(
+	ctx context.Context,
+	logger *log.Entry,
+	cluster *api.Cluster,
+) error {
+	if !z.Supports(cluster) {
+		return ErrProviderNotSupported
+	}
+
+	logger.Infof(
+		"Decommissioning EKS cluster: %s (%s)",
+		cluster.Alias,
+		cluster.ID,
+	)
+
+	awsAdapter, err := z.setupAWSAdapter(logger, cluster)
+	if err != nil {
+		return err
+	}
+	clusterInfo, err := awsAdapter.GetEKSClusterCA(cluster)
+	if err != nil {
+		return err
+	}
+	caData, err := base64.StdEncoding.DecodeString(
+		clusterInfo.CertificateAuthority,
+	)
+	if err != nil {
+		return err
+	}
+
+	cluster.APIServerURL = clusterInfo.Endpoint
+	tokenSource := eks.NewTokenSource(awsAdapter.session, eksID(cluster.ID))
+
+	return z.decommission(
+		ctx,
+		logger,
+		awsAdapter,
+		tokenSource,
+		cluster,
+		caData,
+	)
+}
+
+func (z *ZalandoEKSModifier) GetPostOptions(
+	adapter awsInterface,
+	cluster *api.Cluster,
+	cloudFormationOutput map[string]string,
+) (*PostOptions, error) {
+	res := &PostOptions{}
+
+	clusterInfo, err := adapter.GetEKSClusterCA(cluster)
+	if err != nil {
+		return nil, err
+	}
+	decodedCA, err := base64.StdEncoding.DecodeString(
+		clusterInfo.CertificateAuthority,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	res.APIServerURL = clusterInfo.Endpoint
```

Comment:
Currently, `api_server_url` for EKS clusters isn't the EKS endpoint, but rather the endpoint of the `zalando-iam-proxy`, i.e. for the current `teapot-eks` cluster it's https://kube-eks-1-zalando-iam-aws-proxy.teapot.zalan.do (`{{.Cluster.LocalID}}-zalando-iam-aws-proxy.{{ .Values.hosted_zone }}` from [here](https://github.com/zalando-incubator/kubernetes-on-aws/pull/7808/files#diff-86e8151404f9fb8ac4f9cc3ab78edfa6f1e9db7b45d4365e519adcffaafea331R12) and [here](https://github.bus.zalan.do/teapot/issues/issues/3667#issuecomment-5499773).)

If we run into issues then I think we should use this value here so that templates use the right endpoint.

Reply:
I don't see the `APIServerURL` being used being used when CLM renders templates: https://github.com/zalando-incubator/cluster-lifecycle-manager/blob/853861f3d4238c726901986751018471eddf3896/provisioner/template.go#L84-L138

So I guess it's fine.

Reply:
That was obviously the wrong code section. But I searched for references of the `APIServerURL` field and didn't find any usage in the template rendering code.